azext_iot/tests/helpers.py

```diff
@@ -204,6 +205,9 @@ def assign_role_assignment(
         )
         sleep(wait)
         tries += 1
+    if not output.success():
+        raise CLIInternalError(f"Failed to assign '{assignee}' the role of '{role}' against scope '{scope}'.")
```

Comment:
On second thought (apologies for the run-around), I'm wondering if we should make this a warning?
 Is it better to:
 - fail the whole test run because of an error in the assignment create for a particular resource, or should we 
 - just let the tests continue and log a warning so we can debug later? 
 
 I'm not sure it even matters, just something I wanted to bring up. I'm fine with either. @vilit1 - do you have any preference?

Reply:
ya a warning should be enough for this case, thanks for the feedback!